src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -79,22 +79,26 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetFirmwareRevisionString
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize, size_t & serialNumLen)
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::GetSerialNumber(char * buf, size_t bufSize)
 {
     CHIP_ERROR err;
-    err = Impl()->ReadConfigValueStr(ImplClass::kConfigKey_SerialNum, buf, bufSize, serialNumLen);
+    size_t serialNumLen = 0; // including null-terminator
```

Comment:
I'm not sure what this comment means.  At least the Linux implementation of `ReadConfigValueStr` puts the length of the string _not_ including the null terminator in `serialNumLen`.

Reply:
fixed